### ISSUE TITLE
Deep nesting in I18n webhook data 

### ISSUE THREAD

**Issue by aleksei-savitski-softswiss** [strapi/strapi#12408]
### Describe the bug

When entry with enabled localisation is update, webhook endpoint receives an object with 3 levels of `localizations` arrays.

For example when 3 locales (en, ru, de) are set up, webhook is called 3 times:
<details>
  <summary>First for edited locale with exhaustive nesting</summary>
  
```json
{
  "id": 1,
  "locale": "en",
  "localizations": [
    {
      "id": 3,
      "locale": "ru",
      "localizations": [
        {
          "id": 1,
          "locale": "en",
          "localizations": [
            {
              "id": 2,
              "locale": "de"
            },
            {
              "id": 3,
              "locale": "ru"
            }
          ]
        },
        {
          "id": 2,
          "locale": "de",
          "localizations": [
            {
              "id": 1,
              "locale": "en"
            },
            {
              "id": 3,
              "locale": "ru"
            }
          ]
        }
      ]
    },
    {
      "id": 2,
      "locale": "de",
      "localizations": [
        {
          "id": 1,
          "locale": "en",
          "localizations": [
            {
              "id": 2,
              "locale": "de"
            },
            {
              "id": 3,
              "locale": "ru"
            }
          ]
        },
        {
          "id": 3,
          "locale": "ru",
          "localizations": [
            {
              "id": 1,
              "locale": "en"
            },
            {
              "id": 2,
              "locale": "de"
            }
          ]
        }
      ]
    }
  ]
}
```
</details>

<details>
  <summary>Once for each other locale without `localizations`</summary>
  
```json
{
  "id": 1,
  "locale": "ru"
}
```

```json
{
  "id": 1,
  "locale": "de"
}
```
</details>

### Steps to reproduce the behavior

1. Setup 3 locales
2. Configure Webhook to listen for updates 
3. Create single type with enabled i18n
4. Create an entry of that type for each locale
5. Update and entry for any local
6. See how many times webhook has been executed after single update and what data has been received for each call

### Expected behavior

I expect for one update Webhook is called once. 

- If updated property is localised the webhook data contains only updated entity without localizations.
  <details>
    <summary>example</summary>
    
  ```json
  {
    "id": 1,
    "locale": "en",
     // no `localizations` property here 
  }
  ```
  </details>
- If updated property is shared across locales, the webhook data contains entity for changed locale with localizations but not deeply nested.
  <details>
    <summary>Example</summary>
    
  ```json
  {
    "id": 1,
    "locale": "en",
    "localizations": [
      {
        "id": 3,
        "locale": "ru",
        // no `localizations` property here 
      },
      {
        "id": 2,
        "locale": "de",
        // no `localizations` property here 
      }
    ]
  }
  ```
  </details>

### System

- Node.js version: v14
- NPM version: yarn 1
- Strapi version: 4.0.7
- Database: sqlite
- Operating system: macOS

**Comment by aleksei-savitski-softswiss** [comment]
This deep nesting for localised content types also affects Content Manager API. 

I have a content type with json attribute. It stores keys and strings for translations. The json is quite big. For single locale it has 800 keys and is ~50kb. 

It takes time to open a page with this content type in Strapi UI because the entity is loaded with 3 levels `localizations` arrays, exactly the same as for webhooks in the previous message. For each locale it has the 50kb json, and as I have 12 locales the final response is **20Mb**)

**Comment by dkryvi** [comment]
+

**Comment by aleksei-savitski-softswiss** [comment]
The issue has been fixed with #12758

**Comment by bhavesh-chaudhari** [comment]
This issue still exists in strapi v4.12.0
